types/src/genesis/mod.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::SectorSize;
+use address::Address;
+use num_bigint::bigint_ser;
+use serde::Serialize;
+use vm::TokenAmount;
+
+#[derive(Serialize)]
+enum ActorType {
+    // Account,
+// MultiSig,
+}
+
+#[derive(Serialize)]
+pub struct Actor {
+    actor_type: ActorType,
+    #[serde(with = "bigint_ser")]
+    token_amount: TokenAmount,
+}
+
+#[derive(Serialize)]
+pub struct Miner {
+    owner: Address,
+    worker: Address,
+    peer_id: Vec<u8>,
+
+    #[serde(with = "bigint_ser")]
+    market_balance: TokenAmount,
+    #[serde(with = "bigint_ser")]
+    power_balance: TokenAmount,
+    sector_size: SectorSize,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct Template {
+    accounts: Vec<Actor>,
+    miners: Vec<Miner>,
+    network_name: String,
+    // timestamp: SystemTime,
```

Comment:
these fields should probably be public (no reason not to for something like this). This also applies to the structures above as well

Reply:
made public